jupyter_book_to_htmlbook/text_processing.py

```diff
@@ -41,19 +41,20 @@ def clean_chapter(chapter, rm_numbering=True):
 
 def move_span_ids_to_sections(chapter):
     """
-    Takes span tags with "sec-" ids and moves the id to the parent section tag
-    so Atlas can find the cross reference.
+    Takes empty span tags under a section parent with a heading next sibling
+    and moves the id to the parent section tag so Atlas can find the cross
+    reference.
     """
-    sec_spans = chapter.find_all("span", id=re.compile('sec-*'))
-    for span in sec_spans:
-        # get id
-        span_id = span['id']
-        # get next heading
-        section = span.find_parent('section')
-        # add span id to section
-        section['id'] = span_id
-        # remove span so no dup ids
-        span.decompose()
+    empty_id_spans = chapter.find_all("span", id=True, string="")
+    for span in empty_id_spans:
+        if (
+                span.parent.name == "section" and
+                span.next_sibling.name in ["h1", "h2", "h3", "h4", "h5"]
+           ):
+            # add span id to section
+            span.parent['id'] = span['id']
```

Comment:
Question: Does this overwrite any existing `id` on the parent `section`? And if so, is that the desired behavior?

For example, in the test example, is `id="analytics"` replaced with `id="chp-1"`. I'm guessing that `analytics` is auto-generated from the heading text and is not used by the cross-references, but wanted to confirm.

Reply:
Correct; the span ID overrides the one generated by the jupyter book script based on the title text. In this case (slightly modified from the actual source for posterity), the author explicitly added `chp-1` as a URI (via:
```
(chp-1)=
# Analytics
```
And so my assumption in these cases is that the explicitly named URI is the correct one.